README
LiteCommerce is an open-source ecommerce solution

The mission of our company is to make the Internet convenient and productive business environment for us, our customers and customers of our customers.

We are looking for contributors who could help us extend LiteCommerce and translate it to other languages.
Please report any bugs you find at http://bt.litecommerce.com/

Facebook page: http://www.facebook.com/litecommerce


src/CHANGELOG
Version 2.2. build 39

* SECURITY *
[+] 10 Oct 2008, Arsitov - Improvement (0052998): Subdirectories permissions check has been added. 
[+] 26 Sep 2008, Arsitov - Improvement (0048002): Captcha protection functionality was added. 
[+] 23 Sep 2008, Arsitov - Improvement (0051598): Admin IP protection functionality was added. 
[+] 18 Sep 2008, Alien - Improvement (0026318): Auth code is now required to enter safe mode and to cleanup cache.
[+] 08 Jan 2008, Alien - Improvement (0024451): Admin forms protection functionality was added. 

* PURCHASE/CHECKOUT *
[+] 15 Jul 2008, Joy - Improvement (0036614): The "Redirect customer to cart when adding a product" option now influences redirecting even when the store operates in checkout desk mode.
[+] 16 Oct 2008, Arsitov - Improvement (0053527): eCheck information is now included into admin order notification message when offline eCheck payment method is used.
[+] 17 Oct 2008, Arsitov - Improvement (0053569): A 'Display check number for eCheck payment method' option has been added. 
[+] 17 Oct 2008, Arsitov - Improvement (0046942): Customer notes field has been added to the checkout confirmation page for all basic payment methods.
[!] 14 Nov 2008, Sheriff - Bug (0055176): When shipping charge calculation rules were based on the price of items placed in the shopping cart, the items with the 'free shipping' option were not ignored during shipping calculation process. Fixed. 

* ORDERS MANAGEMENT * 
[+] 16 Jan 2008, Alien - Improvement (0033478): 'Free shipping' is now displayed instead of blank space in the order invoices 'Delivery' field.
[!] 04 Jun 2008, Jasper - Bug (0043916): Disabling a shipping module after an order has been placed, caused an error in the order details section. Fixed. 
[!] 06 Oct 2008, Vgv - Bug (0030188): A grammar mistake in order placement progress message has been fixed.
[!] 13 Oct 2008, Arsitov - Bug (0028534): It was impossible to view orders, created using shipping methods that have been disabled. Fixed.

* STOCK MANAGEMENT * 
[+] 16 Oct 2008, Arsitov - Improvement (0053430): 'Keep existing field default values for product extra fields' feature has been implemented.
[!] 27 Feb 2008, Alien - Bug (0036454): Store administrator was logged off and redirected to the login page when he tried to update some products. Fixed.

* TAXES *
[+] 22 Oct 2008, Arsitov - Improvement (0031994) Bulk tax editing validation has been improved.
[+] 28 Jul 2008, Jasper - Improvement (0029612): A note that defining a a value for 'Tax' is mandatory, has been added to the interface.
[!] 19 Dec 2007, Alien - Bug (0031184): If taxes contained a "=" symbol, a fatal error occurred. Fixed. 
[!] 19 Dec 2007, Alien - Bug (0031998): Recursive tax formulas could be entered in the tax calculator. Fixed. 
[!] 26 Sep 2007, Alien - Bug (0026618): There was a mistake in the tax calculation formula. Fixed.

* SHIPPING*
[!] 05 Jun 2008, Jasper - Bug (0041591): Shipping charges and "Delivery" dropdown were not displayed to customers who were not logged in. Fixed. 

* MEMBERSHIP *
[!] 11 Dec 2007, Alien - Bug (0030012): If membership name contained an apostrophe, an error occurred. Fixed.
[!] 25 Oct 2007, Alien - Bug (0028147): Membership levels were sorted incorrectly. Fixed

* USERS MANAGEMENT * 
[+] 26 Jun 2008, Jasper - Improvement (0028147): It is now impossible to delete administrator's account from the customer zone.
[!] 4 Oct 2007, Alien - Bug (0027060): 'Modify address information' functionality did not copy billing address to shipping address. Fixed

* EXPORT/IMPORT, DB BACKUP *
[+] 03 Oct 2008, Vgv - Improvement (0052527): Extra Fields importing process has been improved.
[+] 08 Oct 2008, Arsitov - Improvement (0052967): Catalog import process has been improved. 
[+] 09 Oct 2008, Arsitov - Improvement (0032316) Import users process has been improved. 
[+] 17 Oct 2008, Arsitov - Improvement (0053474): Previously displayed PHP warning messages are now not displayed during HTML design import/export.
[+] 24 Dec 2007, Alien - Improvement (0032311): Users import compatibility has been improved. 
[!] 04 Jun 2008, Jasper - Bug (0042855): The resulting file of the order export to MS Excel XP/XML caused an error, when it was being opened in MS Excel. Fixed. 
[!] 15 Oct 2008, Arsitov - Bug (0053342): Incorrect message was displayed during database exporting. Fixed. 
[!] 23 Oct 2008, Arsitov - Bug (0054006): The 'ship to' fields were exported incorrectly. Fixed. 

* INSTALLATION *
[+] 27 Oct 2008, Arsitov - Improvement (0052516): Displaying error messages during installation has been improved.
[+] 03 Oct 2008, Arsitov - Improvement (0038495): Warning messages about unstable versions of MySQL have been added to the installer.
[+] 03 Oct 2008, Arsitov - Improvement (0052530): Error messaging about Disabled functions in the installer has been improved. 
[+] 03 Oct 2008, Arsitov- Improvement (0052500): open_basedir - Curl/OpenSSL warning messages have been added in the installer has been improved. 

* CODE REWRITTEN/ELIMINATED *

[+] 17 Oct 2008, Arsitov - Improvement (0031749): Upload file validator and error messaging have been improved.
[+] 09 Oct 2008, Arsitov - Improvement (0053020) Working with images directory has been improved.
[+] 08 Oct 2008, Arsitov - Improvement (0031218): Upgrade process has been improved. 
[+] 06 Oct 2008, Arsitov - Improvement (0030543): Files are now handled in a more flexible way, file permissions now depend on PHP working modes. 
[+] 02 Oct 2008, Arsitov - Improvement (0052435): Memory limit is now dynamically increased during importing/exporting HTML design.
[+] 01 Oct 2008, Arsitov - Improvement (0052265): Working with PHP v.5.2.5 and better has been improved. 
[+] 19 Jun 2008, Jasper - Improvement (0038604): The value of memory_limit is now checked to prevent a situation when a greater value was overwritten with a smaller one. 
[+] 29 Oct 2008, Arsitov - Improvement (0054430): PHPMailer 1.73 has been added. 

src/batch.php
#!/usr/local/bin/php -q -d register_argc_argv=On
<?php
/*
+------------------------------------------------------------------------------+
| LiteCommerce                                                                 |
| Copyright (c) 2003-2009 Creative Development <info@creativedevelopment.biz>  |
| All rights reserved.                                                         |
+------------------------------------------------------------------------------+
| PLEASE READ  THE FULL TEXT OF SOFTWARE LICENSE AGREEMENT IN THE  "COPYRIGHT" |
| FILE PROVIDED WITH THIS DISTRIBUTION.  THE AGREEMENT TEXT  IS ALSO AVAILABLE |
| AT THE FOLLOWING URLs:                                                       |
|                                                                              |
| FOR LITECOMMERCE                                                             |
| http://www.litecommerce.com/software_license_agreement.html                  |
|                                                                              |
| FOR LITECOMMERCE ASP EDITION                                                 |
| http://www.litecommerce.com/software_license_agreement_asp.html              |
|                                                                              |
| THIS  AGREEMENT EXPRESSES THE TERMS AND CONDITIONS ON WHICH YOU MAY USE THIS |
| SOFTWARE PROGRAM AND ASSOCIATED DOCUMENTATION THAT CREATIVE DEVELOPMENT, LLC |
| REGISTERED IN ULYANOVSK, RUSSIAN FEDERATION (hereinafter referred to as "THE |
| AUTHOR")  IS  FURNISHING  OR MAKING AVAILABLE TO  YOU  WITH  THIS  AGREEMENT |
| (COLLECTIVELY,  THE "SOFTWARE"). PLEASE REVIEW THE TERMS AND  CONDITIONS  OF |
| THIS LICENSE AGREEMENT CAREFULLY BEFORE INSTALLING OR USING THE SOFTWARE. BY |
| INSTALLING,  COPYING OR OTHERWISE USING THE SOFTWARE, YOU AND  YOUR  COMPANY |
| (COLLECTIVELY,  "YOU")  ARE ACCEPTING AND AGREEING  TO  THE  TERMS  OF  THIS |
| LICENSE AGREEMENT. IF YOU ARE NOT WILLING TO BE BOUND BY THIS AGREEMENT,  DO |
| NOT  INSTALL  OR USE THE SOFTWARE. VARIOUS COPYRIGHTS AND OTHER INTELLECTUAL |
| PROPERTY  RIGHTS PROTECT THE SOFTWARE. THIS AGREEMENT IS A LICENSE AGREEMENT |
| THAT  GIVES YOU LIMITED RIGHTS TO USE THE SOFTWARE AND NOT AN AGREEMENT  FOR |
| SALE  OR  FOR TRANSFER OF TITLE. THE AUTHOR RETAINS ALL RIGHTS NOT EXPRESSLY |
|                                                                              |
| The Initial Developer of the Original Code is Creative Development LLC       |
| Portions created by Creative Development LLC are Copyright (C) 2003 Creative |
| Development LLC. All Rights Reserved.                                        |
+------------------------------------------------------------------------------+
*
* LiteCommerce ASP Edition Control Center batch processor.
*
* $Id: batch.php,v 1.15 2009/03/17 09:57:35 fundaev Exp $
*
* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: 
*/

// ================== Batch script configuration ==========================

$cfg['host'] = ''; // Server host name where batch is running. Must be equal to 
                   // host name you were using during ASP installation.
$cfg['user'] = ''; // Control Center admin username (email)
$cfg['password'] = ''; // Control Center admin password.
$cfg['mysql_root_user'] = '';       // Privileged user to access MySQL database
$cfg['mysql_root_password'] = '';   // Use saved if empty.
$cfg['access-policy'] = 'web based'; // The default access policy.
$cfg['modules'] = ''; // The default modules set, comma delimited

// ========================================================================

// install/update/remove shop {{{

!isset($_SERVER["REQUEST_METHOD"]) or die("ACCESS DENIED");

// read and parse command line arguments
require_once "./includes/functions.php";
require_once "./includes/decoration.php";
$lib_dir = func_is_php5() ? "lib5" : "lib";
$include_path = ini_set("include_path", $lib_dir);
require_once "System.php";
$whoami = $argv[0];

$argc > 1 or usage();

$con  = new Console_Getopt;
$args = $con->readPHPArgv();
$shortopt = "u:s::p::a::m::d::N::U::C::P::l::z::r::q?";
$longopt = array("url=", "secure_url==", "path=", "access_policy==", "modules==", "database_name==", "create_database==", "database_user==", "create_user==", "database_passwd==", "login==", "password==", "admin_login==", "admin_passwd==", "db_root_user==", "db_root_passwd==", "remove_shop==", "quiet", "help");

@array_shift($args);
$options = $con->getopt($args, $shortopt, $longopt);
if (PEAR::isError($options)) {
    usage($options);
}

// Fill request properties

$_SERVER["HTTP_HOST"] = $cfg['host'];
// install request
$_REQUEST["target"] = "shops";
$_REQUEST["action"] = "install";
// auto-login admin username and password
$_POST["login"] = $cfg['user'];
$_POST["password"] = $cfg['password'];
if (!empty($cfg['access-policy'])) {
    $_POST["shop_profile"] = $cfg['access-policy'];
}
if (!empty($cfg['mysql_root_user'])) {
    $_POST["root_user"] = $cfg['mysql_root_user'];
}    
if (!empty($cfg['mysql_root_password'])) {
    $_POST["root_password"] = $cfg['mysql_root_password'];
}    
if (!empty($cfg['modules'])) {
    $_POST["shop_modules"] = explode(',', $cfg['modules']);    
}

$_POST["shop_db_database_usage"] = "exists";
$_POST["shop_db_user_usage"] = "exists";

// parse command line
foreach ($options[0] as $opt) {
    $param = $opt[1];
	if ($param{0} == '=') $param = substr($param, 1);
    switch($opt[0]) {
        case 'u':
        case '--url':
            $_POST["shop_url"] = $param;
            break;
        case 's':
        case '--secure_url':
            $_POST["shop_secure_url"];
            break;
        case 'p':
        case '--path':
            $_POST["shop_path"] = $param;
            break;
        case 'a':
        case '--access_policy':
            $_POST["shop_profile"] = $param;
            break;
        case 'm':
        case '--modules':
            $_POST["shop_modules"] = explode(',', $param);
            break;
        case 'd':
        case '--database_name':
            $_POST["shop_db_database"] = $param;
            break;
        case 'N':
        case '--create_database':
            $_POST["shop_db_database_usage"] = $param!='Y' ? "exists" : "";
            break;
        case 'U':
        case '--database_user':
            $_POST["shop_db_user"] = $param;
            break;
        case 'C':
        case '--create_user':
            $_POST["shop_db_user_usage"] = $param!='Y' ? "exists" : "";
            break;
        case 'P':
        case '--database_passwd':
            $_POST["shop_db_password"] = $param;
            break;
        case 'l':
        case '--login':
            $_POST["shop_user"] = $param;
            break;
        case 'z':
        case '--password':
            $_POST["shop_password"] = $_POST["shop_password_confirm"] = $param;
            break;
        case '--admin_login':
            $_POST["login"] = $param;
            break;
        case 'z':
        case '--admin_passwd':
            $_POST["password"] = $param;
            break;
        case '--db_root_user':
            $_POST["root_user"] = $param;
            break;
        case 'z':
        case '--db_root_passwd':
            $_POST["root_password"] = $param;
            break;
        case 'r':
        case '--remove_shop':
            $_REQUEST["action"] = "uninstall";
            $_POST["remove_files"] = "Y";
            $_POST["remove_database"] = "Y";
            break;
        case 'q':
        case '--quiet':
            $quiet = true;
            break;
        case '?':
        case '--help':
            usage();
            break;
    }                                               
}

function usage($options = null) {
    global $whoami;
    if (PEAR::isError($options)) {
        print "Error: " . $options->getMessage() . "\n\n";
    }
    $usage =<<<EOT
LiteCommerce ASPE Batch Processor v. 2.2

SYNOPSIS
	batch.php --admin_login=<value> --admin_passwd=<value> [--OPTIONS=<args>]
	batch.php --admin_login=<value> --admin_passwd=<value> -r -u=<value>

OPTIONS
Administrator credentials:
 --admin_login=...          LiteCommerce ASPE administrator login
 --admin_passwd=...         LiteCommerce ASPE administrator password
 --db_root_user=...         MySQL root username (uses saved if empty)
 --db_root_passwd=...       Password for MySQL root user (uses saved if empty)

Shop parameters:
 -u=, --url=...             URL of a new shop, including web directory
                            Example: http://www.example.com/shop1
 -s=, --secure_url=...      HTTPS URL of the new shop (optional)
                            Example: https://secure.example.com/shop1
 -p=, --path=...            Absolute file path to the new shop
 -d=, --database_name=...   MySQL database name for the new shop
 -N=, --create_database=... 'Y' if the database should be created
 -U=, --database_user=...   MySQL database username for the new shop
 -C=, --create_user=...     If this parameter has 'Y' value the user will be created
 -P=, --database_passwd=... Password for MySQL database username
 -l=, --login=...           E-mail of the administrator of the new shop
 -z=, --passwd=...          Password for the administrator of the new shop
 -a=, --access_policy=...   Access policy to assign to the new shop
 -m=, --modules=...         Comma-separated list of modules to install
                            on the new shop (optional)

Operation modes:
 -r, --remove_shop          Uninstall a shop

Other:
 -q, --quiet                Operate in quiet mode
 -?, --help                 Display this help and exit

EOT;
    print $usage;
    exit(1);
}
 
ini_set("include_path", $include_path);

// defines interface
$GLOBALS["XLITE_SELF"] = "cpanel.php";

// reads configuration
include_once "includes/prepend.php";
func_read_classes('modules/asp');

// fires up and runs the LiteCommerce admin interface

if ($quiet) {
    ob_start();
}

$xlite =& func_new("AspXLite");
func_add_decorator('Module', 'AspCPModule');
$xlite->set("aspZone", true);
$xlite->initFromGlobals();
$xlite->set("layout.skin", "admin");
$xlite->run();
$xlite->done();

if ($quiet) {
    ob_end_clean();
}

exit(isset($retcode) ? $retcode : 0);

// }}}

// WARNING :
// Please ensure that you have no whitespaces / empty lines below this message.
// Adding a whitespace or an empty line below this line will cause a PHP error.
?>
